src/d3a/d3a_core/sim_results/file_export_endpoints.py

```diff
@@ -370,50 +371,83 @@ def _accumulate_energy(self, area, child, is_cep: bool):
                     if trade.buyer is child.name:
                         self._total_energy += trade.offer.energy
 
-    def _export_house_pv_self_consumption(self, area):
-        house_pv_device = list()
-        house_load_device = list()
-        trade_by_pv = 0
-        total_energy_bought = 0
-        traded_from_pv = 0
-        if not _is_house_node(area):
-            return
+    def _accumulate_devices(self, area):
         for child in area.children:
-            if isinstance(child.strategy, PVStrategy):
-                house_pv_device.append(child.name)
-            elif _is_load_node(child) or _is_prosumer_node(child):
-                house_load_device.append(child.name)
+            if _is_producer_node(child):
+                self.producer_list.append(child.name)
+            elif _is_load_node(child):
+                self.consumer_list.append(child.name)
+                self.consumer_area_list.append(child.parent)
+                for time, energy in child.strategy.state.desired_energy_Wh.items():
+                    self.total_energy_demanded += energy
+            elif _is_prosumer_node(child):
+                self.ess_list.append(child.name)
+
+            if child.children:
+                self._accumulate_devices(child)
+
+    def _accumulate_self_consumption(self, trade):
+        if trade.seller_origin in self.producer_list and trade.buyer_origin in self.consumer_list:
+            self.total_self_consumption += trade.offer.energy * 1000
```

Comment:
Can you please explain, why you are converting to Wh here (and everywhere)?
Is it because of the loads' desired_energy_Wh ? If true, please keep it, but please append a unit to all energy variables (e.g. `_Wh` Thanks!